jerry-core/ecma/builtin-objects/ecma-builtin-generator-prototype.c

```diff
@@ -83,7 +109,14 @@ ecma_builtin_generator_prototype_object_do (ecma_value_t this_arg, /**< this arg
     return ecma_create_iter_result_object (ECMA_VALUE_UNDEFINED, ECMA_VALUE_TRUE);
   }
 
-  executable_object_p->frame_ctx.call_operation = yield_mode;
+  if (resume_mode == ECMA_GENERATOR_RETURN)
+  {
+    executable_object_p->frame_ctx.byte_code_p = ecma_builtin_generator_prototype_return;
+  }
+  else if (resume_mode == ECMA_GENERATOR_THROW)
+  {
+    executable_object_p->frame_ctx.byte_code_p = ecma_builtin_generator_prototype_throw;
+  }
```

Comment:
Is it guaranteed here that only these two cases can occur? Shouldn't we check the `NEXT` case also here?

Reply:
No need to do anything in the next case. And the enum has no more values.